expected/wasm32-wasi/posix/defined-symbols.txt

```diff
@@ -199,22 +200,17 @@ __sinl
 __small_printf
 __stderr_FILE
 __stderr_used
-__stderr_used
```

Comment:
I could but that's sort of a secondary concern right now. Things that need to get figured out:
- [ ] how to lock `preopens.c` (see my Zulip question), 
- [ ] how best to expose MUSL's `thrd_sleep` instead of cloud ABI's (https://github.com/WebAssembly/wasi-libc/pull/303#discussion_r923921656)
- [ ] where to define, e.g., `__wasilibc_pthread_self` (https://github.com/WebAssembly/wasi-libc/pull/303#discussion_r923920481)
- [ ] remove uses thread cancellation (https://github.com/WebAssembly/wasi-libc/pull/303#discussion_r923918195)
- [ ] decide whether `__get_tp` should even be a defined symbol in single-threaded model (https://github.com/WebAssembly/wasi-libc/pull/303#discussion_r923923837)

Reply:
1. For pre-opens yes, its seems reasonable to use fairly coarse grain locking, at least initially.
2. Seems like we can just remove the `thrd_sleep` alias from `libc-bottom-half/cloudlibc/src/libc/time/nanosleep.c` and start compiling `libc-top-half/musl/src/thread/thrd_sleep.c` which should work for both posix and single threaded builds (that can be its one change if you are looking to pieces of this to land right away).
3. `/libc-top-half/musl/src/thread/pthread_self.c` seems like a reasonable place to define this global.
4. Seems reasonable to skip over thread cancellation initially.  No need to add any calls to the bottom half, and for the top half we can just stub out the impl.
5. I think its reasonable to define `__get_tp` to just always returns a constant in the single threaded build.
6. Indeed, it looks like we should maybe just skip building the threaded build for clang v10 and only build it for recent clang.  The CI only builds with clang v10 to tests that we work with older clang, and I think its reasonable for the threaded build to require a more recent clang.

Reply:
> 3. `/libc-top-half/musl/src/thread/pthread_self.c` seems like a reasonable place to define this global.

Can you show me a place that demonstrates the right way to do this for the WebAssembly target? I'm just not super familiar with how to map a WebAssembly global directly in C...

Reply:
You have to use inline assembly, or in a separate `.s` assembly file.  We recently starting doing it in emscripten here:

https://github.com/emscripten-core/emscripten/blob/517cc860e204b8e659c41649e5542d534bc5cad8/system/lib/pthread/emscripten_tls_init.c#L34-L40